sdk/tables/azure-data-tables/src/tables_clients.cpp

```diff
@@ -0,0 +1,985 @@
+    const std::string& serviceUrl,
+    std::shared_ptr<Azure::Data::Tables::Credentials::SharedKeyCredential> credential,
+    const TableClientOptions& options)
+    : m_url(Azure::Core::Url(serviceUrl))
+{
+  std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> perRetryPolicies;
+  std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> perOperationPolicies;
+  perRetryPolicies.emplace_back(std::make_unique<SwitchToSecondaryPolicy>(
+      m_url.GetHost(), options.SecondaryHostForRetryReads));
+  perRetryPolicies.emplace_back(std::make_unique<TimeoutPolicy>());
+  perOperationPolicies.emplace_back(
+      std::make_unique<ServiceVersionPolicy>(options.ApiVersion.ToString()));
+  m_pipeline = std::make_shared<Azure::Core::Http::_internal::HttpPipeline>(
+      options,
+      _detail::TablesServicePackageName,
+      _detail::ApiVersion,
+      std::move(perRetryPolicies),
+      std::move(perOperationPolicies));
+
+  TableClientOptions newOptions = options;
+  newOptions.PerRetryPolicies.emplace_back(
+      std::make_unique<Azure::Data::Tables::_detail::Policies::SharedKeyLitePolicy>(credential));
+
+  std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> perRetryPolicies2;
+  std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>> perOperationPolicies2;
+  perRetryPolicies2.emplace_back(std::make_unique<SwitchToSecondaryPolicy>(
+      m_url.GetHost(), newOptions.SecondaryHostForRetryReads));
+  perRetryPolicies2.emplace_back(std::make_unique<TimeoutPolicy>());
+  perOperationPolicies2.emplace_back(
+      std::make_unique<ServiceVersionPolicy>(newOptions.ApiVersion.ToString()));
+  m_pipeline = std::make_shared<Azure::Core::Http::_internal::HttpPipeline>(
+      newOptions,
+      _detail::TablesServicePackageName,
+      _detail::ApiVersion,
+      std::move(perRetryPolicies2),
+      std::move(perOperationPolicies2));
+}
+
+TableServicesClient TableServicesClient::CreateFromConnectionString(
+    const std::string& connectionString,
+    const TableClientOptions& options)
+{
+  auto parsedConnectionString = ParseConnectionString(connectionString);
+  auto tablesUrl = std::move(parsedConnectionString.TableServiceUrl);
+
+  if (parsedConnectionString.KeyCredential)
+  {
+    return TableServicesClient(
+        tablesUrl.GetAbsoluteUrl(), std::move(parsedConnectionString.KeyCredential), options);
+  }
+  else
+  {
+    return TableServicesClient(options);
+  }
+}
+
+Azure::Response<Models::PreflightCheckResult> TableServicesClient::PreflightCheck(
+    Models::PreflightCheckOptions const& options,
+    Core::Context const& context)
+{
+  auto url = m_url;
+  url.AppendPath(options.TableName);
+  Core::Http::Request request(Core::Http::HttpMethod::Options, url);
+  request.SetHeader("Origin", options.Origin);
+  request.SetHeader("Access-Control-Request-Method", Core::Http::HttpMethod::Options.ToString());
+  auto rawResponse = m_pipeline->Send(request, context);
```

Comment:
From https://github.com/Azure/azure-sdk-for-cpp/pull/5246#discussion_r1448174411
> An alternative stop-gap solution to adding this in core, is to just use the "OPTIONS" string. HttpMethod is an extensible enum which supports constructing from any string and is designed exactly for such scenarios.

You could just use `Core::Http::HttpMethod("OPTIONS")` here in these two places, and not have to be worried about adding new surface area to core and GA it, just to beta Tables.